src/components/TextField.tsx

```diff
@@ -39,19 +45,38 @@ export const TextField: FC<FieldProps<TextFieldSchema>> = ({
     renderAfter,
     disabled,
     readOnly,
+    copyToClipboard,
+    validateOnChange,
+    loadingValidate,
   } = field;
 
   const { isReadOnly } = useContext(Ctx);
 
   const fieldStyles = useStyles<FieldStyles>('textField', styles);
 
-  const { register, control } = useFormContext();
+  const { register, control, watch } = useFormContext();
 
   const errorMessage = useErrorMessage(name, label);
 
   const values = useWatch({
     control,
   });
+  const value = watch(name);
+  const debouncedValue = useDebounce(value, 500);
+
+  const [valid, setValid] = useState(true);
+
+  useEffect(() => {
+    const validate = async () => {
```

Comment:
What happens if `validateOnChange` is undefined?

Reply:
Absolutely nothing haha, `valid` will still be true and the icons are shown if `validateOnChange` is defined anyway

Reply:
Should we wrap it all if an `if validateOnChange`?